### PR TITLE
[ARCTIC-1494][AMS] table optimize history abnormal display

### PR DESCRIPTION
**Pull request by xujiangfeng001**
## Why are the changes needed?

fix https://github.com/NetEase/arctic/issues/1494

## Brief change log
Modified SQL for listing successful optimizing history to avoid reading records with the same name but not belonging to this table.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by zhoujinsong** [comment]
Thanks a lot for your contribution!

However, I still want to encourage you to provide some key details of the code changes in the `Brief change log` section of your PR description to help reviewers better understand your code.

For example, for this PR, the change log could be "Modified SQL for listing successful optimizing history to avoid reading records with the same name but not belonging to this table."

**Comment by shidayang** [comment]
Thanks for you contribution. 
You need to make this PR more compliant with the standards.  The. title need rename to "[ARCTIC-1494][AMS] table optimize history abnormal display"， some other see https://mp.weixin.qq.com/s/mpR9jREdBV21bFHxtdlAmw

File: ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/OptimizingMapper.java [hunk 1]
```diff
@@ -89,11 +89,13 @@ List<TableOptimizingProcess> selectOptimizingProcessesByTable(
...
 
-  @Select("SELECT process_id, table_id, catalog_name, db_name, table_name, target_snapshot_id," +
-      " target_change_snapshot_id, status," +
-      " optimizing_type, plan_time, end_time, fail_reason, summary FROM table_optimizing_process" +
-      " WHERE catalog_name = #{catalogName} AND db_name = #{dbName} AND table_name = #{tableName}" +
-      " AND status = 'SUCCESS'")
+  @Select("SELECT a.process_id, a.table_id, a.catalog_name, a.db_name, a.table_name, a.target_snapshot_id," +
+      " a.target_change_snapshot_id, a.status, a.optimizing_type, a.plan_time, a.end_time," +
+      " a.fail_reason, a.summary FROM table_optimizing_process a" +
+      " LEFT JOIN table_identifier b ON a.table_id = b.table_id" +
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/OptimizingMapper.java:95]: This place may be inner join?

**Review by shidayang — CHANGES_REQUESTED** [review]

File: ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/OptimizingMapper.java (hunk 1 above)
Comment (xujiangfeng001) [reply ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/OptimizingMapper.java:95]: This is my negligence, I will immediately modify and resubmit the code.

**Comment by xujiangfeng001** [comment]
> Thanks for you contribution. You need to make this PR more compliant with the standards. The. title need rename to "[ARCTIC-1494][AMS] table optimize history abnormal display"， some other see https://mp.weixin.qq.com/s/mpR9jREdBV21bFHxtdlAmw

Thank you very much for your suggestion.  I will pay special attention next time.

**Review by shidayang — APPROVED** [review]

**Review by shidayang — APPROVED** [review]

**Comment by shidayang** [comment]
LGTM

**Pull request merged**